distributed/tests/test_s3.py

```diff
@@ -27,59 +27,112 @@
 
 
-def test_get_list_of_summary_objects():
-    L = get_list_of_summary_objects(test_bucket_name, prefix='test/accounts',
-                                    anon=True)
+def test_get_list_of_summary_objects(s3):
+    L = s3.ls(test_bucket_name + '/test')
 
     assert len(L) == 2
-    assert list(map(lambda o: o.key, L)) == sorted(list(files))
+    assert [l.lstrip(test_bucket_name).lstrip('/') for l in sorted(L)] == sorted(list(files))
 
-    L2 = get_list_of_summary_objects('s3://' + test_bucket_name, prefix='/test/accounts',
-                                    anon=True)
+    L2 = s3.ls('s3://' + test_bucket_name + '/test')
 
     assert L == L2
 
 
-def test_read_keys_from_bucket():
+def test_read_keys_from_bucket(s3):
     for k, data in files.items():
-        file_contents = read_content_from_keys('distributed-test', k, anon=True)
-
+        file_contents = s3.cat('/'.join([test_bucket_name, k])) 
         assert file_contents == data
 
-    assert (read_content_from_keys('s3://distributed-test', k, anon=True) ==
-            read_content_from_keys('distributed-test', k, anon=True))
-
+    assert (s3.cat('/'.join([test_bucket_name, k])) ==
+            s3.cat('s3://' + '/'.join([test_bucket_name, k])))
+
+
+def test_seek_delimiter(s3):
+    fn = 'test/accounts.1.json'
+    data = files[fn]
+    with s3.open('/'.join([test_bucket_name, fn])) as f:
+        seek_delimiter(f, b'}', 0)
+        assert f.tell() == 0
+        f.seek(1)
+        seek_delimiter(f, b'}', 5)
+        assert f.tell() == data.index(b'}') + 1
+        seek_delimiter(f, b'\n', 5)
+        assert f.tell() == data.index(b'\n') + 1
+
+
+def test_read_block(s3):
+    import io
+    data = files['test/accounts.1.json']
+    lines = io.BytesIO(data).readlines()
+    path = 'distributed-test/test/accounts.1.json'
+    assert s3.read_block(path, 1, 35, b'\n') == lines[1]
+    assert s3.read_block(path, 0, 30, b'\n') == lines[0]
+    assert s3.read_block(path, 0, 35, b'\n') == lines[0] + lines[1]
+    assert s3.read_block(path, 0, 5000, b'\n') == data
+    assert len(s3.read_block(path, 0, 5)) == 5
+    assert len(s3.read_block(path, 0, 5000)) == len(data)
+    assert s3.read_block(path, 5000, 5010) == b''
 
-def test_list_summary_object_with_prefix_and_delimiter():
-    keys = get_list_of_summary_objects(test_bucket_name, 'nested/nested2/',
-                                       delimiter='/', anon=True)
 
-    assert len(keys) == 2
-    assert [k.key for k in keys] == [u'nested/nested2/file1',
-                                     u'nested/nested2/file2']
+@gen_cluster(timeout=60)
+def test_read_bytes(s, a, b):
+    e = Executor((s.ip, s.port), start=False)
+    yield e._start()
 
-    keys = get_list_of_summary_objects(test_bucket_name, 'nested/', anon=True)
+    futures = read_bytes(test_bucket_name+'/test/accounts.*', lazy=False)
+    assert len(futures) >= len(files)
+    results = yield e._gather(futures)
+    assert set(results).issuperset(set(files.values()))
```

Comment:
Why is this true?  Isn't it possible for the blocks of bytes to be cut differently than the files on S3?